src/services/users_service.ts

```diff
@@ -22,32 +25,100 @@ export const signin = async () => {
 };
 
 export const readAll = async () => {
-  try {
-    console.log('success'); //라우팅 테스트용
-  } catch (err) {}
+  const users = await model.User.findAll();
+  return users;
 };
 
-export const readOne = async () => {
-  try {
-  } catch (err) {}
+export const readOne = async (id: any) => {
+  const user = await model.User.findOne({
+    where: {
+      id,
+    },
+  });
+  return user;
 };
 
 export const readOneByEmail = async () => {
   try {
   } catch (err) {}
 };
 
-export const updateInfo = async () => {
-  try {
-  } catch (err) {}
+export const updateAlarm = async (id: any, isAlarmSet: boolean, alarmTime: Date) => {
+  let userAlarmTime: any;
+  if (alarmTime) {
+    await model.User.update(
+      {
+        isAlarmSet,
+        alarmTime,
+      },
+      {
+        where: {
+          id,
+        },
+      }
+    ).then(async () => {
+      userAlarmTime = await model.User.findOne({
+        attributes: ['isAlarmSet', 'alarmTime'],
+        where: {
+          id,
+        },
+      });
+    });
+
+    return userAlarmTime;
+  } else {
+    await model.User.update(
+      {
+        isAlarmSet,
+      },
+      {
+        where: {
+          id,
+        },
+      }
+    ).then(async () => {
+      userAlarmTime = await model.User.findOne({
+        attributes: ['isAlarmSet', 'alarmTime'],
+        where: {
+          id,
+        },
+      });
+    });
+    return userAlarmTime;
+  }
 };
 
-export const updatePassword = async () => {
-  try {
-  } catch (err) {}
+export const checkPassword = async (user: User, password: string) => {
+  const hashedPassword = crypto.pbkdf2Sync(password, user.passwordSalt, 10000, 64, 'sha512').toString('base64');
+  if (hashedPassword != user.password) {
+    return false;
+  } else {
+    return true;
+  }
 };
 
-export const deleteOne = async () => {
-  try {
-  } catch (err) {}
+export const updatePassword = async (id: any, password: string) => {
+  const salt = crypto.randomBytes(64).toString('base64');
+  const hashedPassword = crypto.pbkdf2Sync(password, salt, 10000, 64, 'sha512').toString('base64'); //채원이 hash service
+  await model.User.update(
+    {
+      password: hashedPassword,
+      passwordSalt: salt,
+    },
+    {
+      where: {
+        id,
+      },
+    }
+  );
+  return;
+};
+
+export const deleteOne = async (id: any) => {
+  await model.User.destroy({
+    where: {
+      id,
+    },
+  });
+  return;
 };
```

Comment:
컨트롤러에서 id에 해당하는 유저를 먼저 찾고, 그유저를 파라미터에 넣어주면

```suggestion
export const deleteOne = async (user: User) => {
  const deletedUser = await user.destroy({
    where: { id },
  });
  return deletedUser;
};
```
이렇게하면 리턴값으로 삭제된 유저의 정보를 줄수있음

Reply:
update, destroy 둘다 이런식으로 처리할 수 있어